python3.8/cookiecutter-aws-sam-hello-python/{{cookiecutter.project_name}}/tests/integration/test_api_gateway.py

```diff
@@ -0,0 +1,40 @@
+import os
+from unittest import TestCase
+
+import boto3
+import requests
+
+"""
+Make sure env variable AWS_SAM_STACK_NAME exists with the name of the stack we are going to test. 
+"""
+
+
+class TestApiGateway(TestCase):
+    api_endpoint: str
+
+    def setUp(self) -> None:
+        """
+        Based on the provided env variable AWS_SAM_STACK_NAME,
+        here we use cloudformation API to find out what the HelloWorldApi URL is
+        """
+        stack_name = os.environ.get("AWS_SAM_STACK_NAME")
+        self.failIf(not stack_name, "Cannot find env var AWS_SAM_STACK_NAME")
+
+        client = boto3.client("cloudformation")
+
+        response = client.describe_stacks(StackName=stack_name)
+        stacks = response["Stacks"]
+        self.assertTrue(stacks, f"Cannot find stack {stack_name}")
```

Comment:
All of this can be abstracted out in a method in TestCase class and probably shared across multiple tests.

Reply:
Extracted